R/http.R

```diff
@@ -352,11 +352,31 @@ httpRCurl <- function(protocol,
     error = function(e, ...) {
       # ignore errors resulting from timeout or user abort
       if (identical(e$message, "Callback aborted") ||
-          identical(e$message, "transfer closed with outstanding read data remaining"))
-        return
+          identical(e$message, "transfer closed with outstanding read data remaining")) {
+        return()
+      }
+
+      # detect errors due to overly-large bundles
+      if (grepl("long vectors not supported yet", e$message, fixed = TRUE)) {
+
+        msg <- "Deployment failed due to overly-large bundle."
+
+        prescription <- if (!identical(getOption("rsconnect.http"), "curl"))
+          "Try setting 'options(rsconnect.http = \"curl\")' and re-deploying your application."
```

Comment:
Note that this option *only* works reliably on OS X (it will probably never work on Windows and would only work on Linux if the user had installed the curl command line utility).